synchronization/src/main/java/de/cyface/synchronization/ConnectionBroadcastReceiver.java

```diff
@@ -0,0 +1,84 @@
+package de.cyface.synchronization;
+
+import static de.cyface.synchronization.CyfaceConnectionListener.SYNC_FINISHED;
+import static de.cyface.synchronization.CyfaceConnectionListener.SYNC_MEASUREMENT_ID;
+import static de.cyface.synchronization.CyfaceConnectionListener.SYNC_POINTS_TO_TRANSMIT;
+import static de.cyface.synchronization.CyfaceConnectionListener.SYNC_POINTS_TRANSMITTED;
+import static de.cyface.synchronization.CyfaceConnectionListener.SYNC_PROGRESS;
+import static de.cyface.synchronization.CyfaceConnectionListener.SYNC_STARTED;
+
+import java.util.Collection;
+import java.util.HashSet;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+
+import de.cyface.utils.Validate;
+
+/**
+ * {@link BroadcastReceiver} for the {@link CyfaceConnectionListener} events.
+ *
+ * @author Armin Schnabel
+ * @version 1.0.0
+ * @since 2.5.0
+ */
+public class ConnectionBroadcastReceiver extends BroadcastReceiver {
+
+    static final String TAG = "de.cyface.sync";
+    /**
+     * The interested parties for synchronization events.
+     */
+    private Collection<ConnectionListener> connectionListener;
+
+    public ConnectionBroadcastReceiver(final Context context) {
+        this.connectionListener = new HashSet<>();
+        final IntentFilter filter = new IntentFilter();
+        filter.addAction(SYNC_FINISHED);
+        filter.addAction(SYNC_PROGRESS);
+        filter.addAction(SYNC_STARTED);
+        context.registerReceiver(this, filter);
+    }
+
+    @Override
+    public void onReceive(final Context context, final Intent intent) {
+        Validate.notNull(intent.getAction());
+        switch (intent.getAction()) {
+            case SYNC_STARTED:
+                long pointsToTransmit = intent.getLongExtra(SYNC_POINTS_TO_TRANSMIT, -1L);
+                Validate.isTrue(pointsToTransmit >= 0L);
+                for (final ConnectionListener listener : connectionListener) {
+                    listener.onSyncStarted(pointsToTransmit);
+                }
+                break;
+            case SYNC_FINISHED:
+                for (final ConnectionListener listener : connectionListener) {
+                    listener.onSyncFinished();
+                }
+                break;
+            case SYNC_PROGRESS:
+                pointsToTransmit = intent.getLongExtra(SYNC_POINTS_TO_TRANSMIT, -1L);
+                final long transmittedPoints = intent.getLongExtra(SYNC_POINTS_TRANSMITTED, -1L);
+                final long measurementId = intent.getLongExtra(SYNC_MEASUREMENT_ID, -1L);
+                Validate.isTrue(pointsToTransmit > 0L);
+                Validate.isTrue(transmittedPoints > 0L);
+                Validate.isTrue(measurementId > 0L);
+
+                for (final ConnectionListener listener : connectionListener) {
+                    if (listener != null) {
```

Comment:
Why is there a null check here, but not in the iterations above? Is it even possible for a listener to become null? Either way; either this check is useless or similar checks are missing for SYNC_FINISHED and SYNC_STARTED.

Reply:
This was old code which I moved. I now dropped the null check. If this can actually occur, I think the SDK user forgot to remove the listener.